### PR TITLE
[WIP] Test cases and minor fixes for "Fix inlining on subtypes"

### PR DESCRIPTION
**Pull request by mpetruska**
This PR contains a test case and adds some minor fixes for PR #763, which contains the fix for issues #614 and #740 .

### CODE REVIEW

**Comment by KevinRansom** [comment]
@mpetruska  we will likely get to look at this in the new year but it sure was a great present, thank you.

File: src/fsharp/Optimizer.fs [hunk 1]
```diff
@@ -3055,7 +3062,7 @@ and OptimizeBinding cenv isRec env (TBind(v,e,spBind)) =
             then {einfo with Info=UnknownValue} 
             else einfo 
         if v.MustInline  && IsPartialExprVal einfo.Info then 
-            errorR(InternalError("the mustinline value '"^v.LogicalName^"' was not inferred to have a known value",v.Range));
+            errorR(NumberedError(FSComp.SR.optValueMarkedInlineButIncomplete(v.DisplayName), v.Range))
```
Comment (mpetruska) [inline src/fsharp/Optimizer.fs]: I'm not entirely sure that this only happens when `valu` (at line 2749) is set to `UnknownValue`,

File: tests/fsharp/optimize/inline/neg01.bsl [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+neg01.fs(16,22,16,24): optimize error FS1113: The value 's2' was marked inline but its implementation makes use of an internal or private function which is not sufficiently accessible
+
+neg01.fs(16,22,16,24): optimize error FS1113: The value 's2' was marked inline but its implementation makes use of an internal or private function which is not sufficiently accessible
```
Comment (mpetruska) [inline tests/fsharp/optimize/inline/neg01.bsl]: Unfortunately I did not find out yet why the error is reported twice.

**Comment by KevinRansom** [comment]
@mpetruska in your opinion is this PR ready to pull?

File: src/fsharp/Optimizer.fs [hunk 3]
```diff
@@ -1066,7 +1066,7 @@ let AbstractLazyModulInfoByHiding isAssemblyBoundary mhi =
            ValInfos = 
                ValInfos(ss.ValInfos.Entries 
                          |> Seq.filter (fun (vref,_) -> not (hiddenVal vref.Deref))
-                         |> Seq.map (fun (vref,e) -> check (* "its implementation uses a binding hidden by a signature" m *)  vref (abstractValInfo e) )) } 
+                         |> Seq.map (fun (vref,e) -> (vref, (abstractValInfo e) ))) }
```
Comment (dsyme) [inline src/fsharp/Optimizer.fs]: What's this change for and which test case covers it?

File: tests/fsharp/core/subtype/test.fsx [hunk 4]
```diff
@@ -1696,6 +1696,25 @@ module RecordPropertyConstraintTests =
     check "ckjwnewk" (f8()) (System.TimeSpan.FromSeconds 2.0) // after mutation
     check "ckjwnewk" (f10()) "Gary"
 
+// See https://github.com/Microsoft/visualfsharp/issues/740 - inlining on subtypes was not allowed
```
Comment (dsyme) [inline tests/fsharp/core/subtype/test.fsx]: This test case looks the same as https://github.com/Microsoft/visualfsharp/pull/763/files#diff-e2959d5dc5cc275246c938d409549ee1R1699, which is added as a succeeding test case.  Have you checked that this test case still gives the errors shown after the fix is made?

**Comment by dsyme** [comment]
Marking WIP until the questions above are answered

**Comment by dsyme** [comment]
I think we can close this since there's been no action for quite a while.  I'm not sure what additional cases are being covered here

**Pull request closed without merging**